qemu/tests/chardev_free_port.py

```diff
@@ -0,0 +1,46 @@
+from virttest import error_context, env_process
+from virttest import utils_misc
+
+from qemu.tests.virtio_console import add_chardev
+
+
+@error_context.context_aware
+def run(test, params, env):
+    """
+    qemu should try to find a free port value by to=<port> with unix socket and tcp options:
+    1) boot guest with socket with -qmp tcp:0:4444.
+    2) telnet localhost.
+    3) check output of {"execute": "qmp_capabilities"}, {"execute":"query-chardev"}.
+    4) repeat step 1 to 3 with -qmp tcp:0:4445 and telnet to port 4445.
+    :param test: QEMU test object
+    :param params: Dictionary with the test parameters
+    :param env: Dictionary with test environment.
+    """
+
+    chardev_infos = []
+    port_nums = []
+    vms = params.get("vms").split()
+    _vm = env.get_vm(vms[0])
+    char_devices = add_chardev(_vm, params)
+    chardev_params = char_devices[0].params
+    for vm_ind, vm in enumerate(vms):
+        if vm_ind == 1:
+            host = chardev_params['host']
+            chardev_to = utils_misc.find_free_ports(int(port_nums[0]) + 1, 6000, 1, host)
+            chardev_params['to'] = str(chardev_to[0])
+        for device in char_devices:
```

Comment:
Keep consistent with L24~L25, only check one chardev in this case, no need to add this loop.
```
char_device = add_chardev(_vm, params)[0]
char_device.params...
char_device.cmdline...
```

Reply:
hi problem there are multiple chardev devices in the vm

Reply:
> hi problem there are multiple chardev devices in the vm
 No, since you use `chardev_params = char_devices[0].params`. keep consistent.